goagen/codegen/types.go

```diff
@@ -852,8 +852,8 @@ func toSlice(val []interface{}) string {
 
 const (
 	mArrayTmpl = `{{$tmp := tempvar}}{{tabs .depth}}{{$tmp}} := make([]{{gonative .elemType.Type}}, len({{.source}}))
-{{tabs .depth}}for i, r := range {{.source}} {
-{{marshalAttribute .elemType (printf "%s[*]" .context) "r" (printf "%s[i]" $tmp) (add .depth 1)}}
+{{$tmpIndex := tempvar}}{{$tmpElement := tempvar}}{{tabs .depth}}for {{$tmpIndex}}, {{$tmpElement}} := range {{.source}} {
+{{marshalAttribute .elemType (printf "%s[*]" .context) (printf "%s" $tmpElement) (printf "%s[%s]" $tmp $tmpIndex) (add .depth 1)}}
```

Comment:
I don't think you need to do the `printf "%s" $tmpElement` here just `$tmpElement` should do it.